libfoundation/src/foundation-foreign.cpp

```diff
@@ -238,6 +280,92 @@ struct pointer_type_desc_t {
     static constexpr auto& hash_func = MCHashPointer;
 };
 
+struct uintsize_type_desc_t: public integral_type_desc_t<size_t>  {
+    static constexpr auto& type_info = kMCUIntSizeTypeInfo;
+    static constexpr auto describe_format = "<foreign unsigned size %zu>";
+};
+struct sintsize_type_desc_t: public integral_type_desc_t<ssize_t>  {
+    static constexpr auto& type_info = kMCSIntSizeTypeInfo;
+    static constexpr auto describe_format = "<foreign signed size %zd>";
+};
+
+struct uintptr_type_desc_t: public integral_type_desc_t<uintptr_t>  {
+    static constexpr auto& type_info = kMCUIntPtrTypeInfo;
+    static constexpr auto describe_format = "<foreign unsigned intptr %zu>";
+};
+struct sintptr_type_desc_t: public integral_type_desc_t<intptr_t>  {
+    static constexpr auto& type_info = kMCSIntPtrTypeInfo;
+    static constexpr auto describe_format = "<foreign signed intptr %zd>";
+};
+
+/**/
+
+struct cbool_type_desc_t: public bool_type_desc_t
+{
+    static constexpr auto& type_info = kMCCBoolTypeInfo;
+};
+struct cchar_type_desc_t: public integral_type_desc_t<char> {
+    static constexpr auto& type_info = kMCCCharTypeInfo;
+    static constexpr auto describe_format = "<foreign c char '%c'>";
```

Comment:
Hmm - I pondered that - the rationale for using 'c' is that in LCB, if we make foreign binding more explicit then:
```
  foreign c handler ...
```
Might be more consistent (with regards token casing rules) as 'c' and not 'C'.